File: docker/nodejs/execution_server.js
const express = require('express');
const { exec } = require('child_process');
const app = express();
const port = 5002;

app.use(express.json());

app.post('/execute', (req, res) => {
  const code = req.body.code || '';
  
  const child = exec(`node -e "${code.replace(/"/g, '\\"')}"`, {
    timeout: 5000
  }, (error, stdout, stderr) => {
    if (error) {
      return res.status(500).json({
        success: false,
        output: error.message || stderr
      });
    }
    
    res.json({
      success: true,
      output: stdout || stderr
    });
  });
});

app.listen(port, () => {
  console.log(`Node.js execution server running on port ${port}`);
});

File: docker/python/execution_server.py
from flask import Flask, request, jsonify
import subprocess
import os

app = Flask(__name__)

@app.route('/execute', methods=['POST'])
def execute():
    code = request.json.get('code', '')
    
    try:
        process = subprocess.run(
            ['python', '-c', code],
            capture_output=True,
            text=True,
            timeout=5
        )
        
        return jsonify({
            'success': True,
            'output': process.stdout or process.stderr
        })
    except subprocess.TimeoutExpired:
        return jsonify({
            'success': False,
            'output': 'Execution timed out'
        }), 408
    except Exception as e:
        return jsonify({
            'success': False,
            'output': str(e)
        }), 500

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5001)